File: src/henaxel/raytracer/materials/ConstantTexture.java
package henaxel.raytracer.materials;

import henaxel.node.BaseNode;
import henaxel.raytracer.utils.Vec3;

import java.awt.*;

public class ConstantTexture implements Texture {
    private Color color;
    
    public ConstantTexture(Color color) {
        this.color = color;
    }
    public ConstantTexture(double r, double g, double b) {
        this.color = new Color((float)r, (float)g, (float)b);
    }
    
    @Override
    public Color color(double u, double v, Vec3 hitPos) {
        return color;
    }

    @Override
    public BaseNode getNode() {
        return null;    //TODO: Implement
    }
}

File: src/main/java/henaxel/materials/ConstantTexture.kt
package henaxel.materials;


import java.awt.*;

class ConstantTexture(private var color: Color): Texture("Constant Texture", null, null, null, null) {

    constructor(r: Double, g: Double, b: Double) : this(Color(r.toFloat(), g.toFloat(), b.toFloat()))
    

    override fun color(u: Double, v: Double): Color {
        return color;
    }
}
